common-files/utils/web3.mjs

```diff
@@ -54,13 +54,25 @@ export default {
     if (!config.WEB3_OPTIONS.from) throw Error('config WEB3_OPTIONS.from is not set');
     if (!config.ETH_PRIVATE_KEY) throw Error('config ETH_PRIVATE_KEY not set');
 
+    const fromAddress = await this.web3.eth.accounts.privateKeyToAccount(config.ETH_PRIVATE_KEY);
+
     const tx = {
+      from: fromAddress.address,
       to: contractAddress,
       data: rawTransaction,
       value,
-      gas: config.WEB3_OPTIONS.gas,
       gasPrice: config.WEB3_OPTIONS.gasPrice,
     };
+    let gas;
+    try {
```

Comment:
This block of code is repeated in other parts, please, create an utility method so that we can reuse it.

Reply:
What I mean is that this block
```
    try {
      gas = await this.web3.eth.estimateGas(tx);
      logger.debug({ msg: 'Gas estimated at', gas });
    } catch (error) {
      gas = config.WEB3_OPTIONS.gas;
      logger.warn({ msg: 'Gas estimation failed, use default', gas });
    }
```
 can be a utility function that can be used everywhere it is needed. e.g.
```
async function estimateGas(tx) {
    try {
      gas = await this.web3.eth.estimateGas(tx);
      logger.debug({ msg: 'Gas estimated at', gas });
    } catch (error) {
      gas = config.WEB3_OPTIONS.gas;
      logger.warn({ msg: 'Gas estimation failed, use default', gas });
    }
    return gas;
}

...
gas = estimateGas(tx);
...
```

Reply:
i don't think this is needed